libsolutil/Dominator.h

```diff
@@ -312,10 +313,15 @@ class DominatorFinder
 
 		// step 4
 		// Compute idom in DFS order
-		idom[0] = 0;
+		// The entry vertex does not have an immediate dominator.
+		idom[0] = std::nullopt;
```

Comment:
Shouldn't that rather be an assert? You initialize the whole array with `nullopt` and if previous steps end up changing that for the root then there's probably something wrong.

Reply:
True, changed.